desc/grid.py

```diff
@@ -199,6 +203,28 @@ def sym(self):
         """bool: True for stellarator symmetry, False otherwise."""
         return self.__dict__.setdefault("_sym", False)
 
+    @property
+    def is_meshgrid(self):
```

Comment:
"meshgrid" is a matlab/numpy term. I think the more mathy term would be "tensor product grid" (but im fine with either)